compiler/rustc_expand/src/proc_macro.rs

```diff
@@ -15,6 +15,7 @@ const EXEC_STRATEGY: pm::bridge::server::SameThread = pm::bridge::server::SameTh
 
 pub struct BangProcMacro {
     pub client: pm::bridge::client::Client<fn(pm::TokenStream) -> pm::TokenStream>,
+    pub name: &'static str,
```

Comment:
I've been removing the macro definition name from these structures for a couple of times already, because the way the macro is expanded shouldn't depend on its original name in any way.

Do you specifically need the definition name, or a name with which the macro was called would be better?

Reply:
For the call-site name you can take `expn_data.kind.descr()` in `proc_macro_server::Rustc::new`.

Reply:
Ah thank you I'll use that instead, the call-site macro name will be fine indeed: I mostly wanted to show which expansions and locations (coming in a future PR once I introduce some way for the self-profiler to only do work recording arguments when argument recording was opted into).